compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -144,20 +144,21 @@ class SyntheticMethods(thisTransformer: DenotTransformer) {
 
     /** The class
      *
+     *   package p
      *    case class C(x: T, y: T)
      *
      *  gets the hashCode method:
      *
      *    def hashCode: Int = {
-     *      <synthetic> var acc: Int = 0xcafebabe;
+     *      <synthetic> var acc: Int = "p.C".hashCode // constant folded
      *      acc = Statics.mix(acc, x);
      *      acc = Statics.mix(acc, Statics.this.anyHash(y));
      *      Statics.finalizeHash(acc, 2)
      *    }
      */
     def caseHashCodeBody(implicit ctx: Context): Tree = {
       val acc = ctx.newSymbol(ctx.owner, "acc".toTermName, Mutable | Synthetic, defn.IntType, coord = ctx.owner.pos)
-      val accDef = ValDef(acc, Literal(Constant(0xcafebabe)))
+      val accDef = ValDef(acc, Literal(Constant(clazz.fullName.toString.hashCode)))
```

Comment:
Neither are the case accessors (they could be `var`s).

I guess a motivation would be to have more stable hash codes for a local case class so that it wasn't volatile wrt fresh name suffixes.